### DISCUSSION TITLE
Oracle DB and Form Blocks

### DISCUSSION THREAD

**Discussion by Sagestum** [Budibase/budibase#10886] [Help]
Hello everyone,

I am trying to create user-friendly tables for my colleagues using data from our internal ticket system. I have successfully connected the Oracle DB to Budibase, and everything works well except for a few limitations in SQL queries, such as Form Blocks. It's a bit disappointing, but I can live with it.

Although the Budibase Oracle Connector doesn't directly display any tables from the Oracle DB, I believe this is due to the lack of support for Oracle views. However, the queries themselves work fine.

Is there a simple way to import data from Oracle SQL into the Budibase internal database? I'm thinking of a solution where the Oracle SQL query (which doesn't take much time) is imported into the Budibase internal database every time someone opens the web app. That could be a good work a round, for my problem. 

Alternatively, are there any better alternatives for achieving this?

I would like to display a Form Block containing the tickets, and when clicked, a new webpage (or side panel) should show detailed text information.

Thank you very much.

Best regards,
Christian

**Comment by melohagan** [comment]
Hey @Sagestum 

One approach could be to explore [Automations](https://docs.budibase.com/docs/automation-steps).

Your flow would look roughly like:

1. Cron job trigger (every minute or less if the Oracle data isn't updated that frequently)
2. External Data Connector -> Return data from your Oracle SQL query
3. Query Rows -> Return all matching rows between your internal Budibase DB and the SQL query
4. JavaScript Block -> Return an object with two arrays (one for the matching rows, one of all the non-matching rows)
5. Update Row -> Loop over each matching and update as required
6. Create Row -> Loop over each non-matching row and create as required

This is fairly complex, a bit awkward to maintain, and possibly not very performant if there's a lot of rows.

The second approach is to use the SQL queries as the source for your forms and tables, where you need to manage the saving, pagination, and default values more manually:

- [Posting form data to a custom query](https://docs.budibase.com/docs/data-sources#tutorial-posting-form-data-to-a-custom-query)
- [Default form field values](https://docs.budibase.com/docs/form-bindings#default-values)
- [Pagination with custom SQL queries](https://github.com/Budibase/budibase/discussions/7313#discussioncomment-3430353)
